cmd/create/machinepool/machinepool.go

```diff
@@ -574,13 +575,33 @@ func parseTaints(taints string) ([]*cmv1.TaintBuilder, error) {
 	if taints == "" {
 		return taintBuilders, nil
 	}
+	var errs []error
 	for _, taint := range strings.Split(taints, ",") {
 		if !strings.Contains(taint, "=") || !strings.Contains(taint, ":") {
-			return nil, fmt.Errorf("Expected key=value:scheduleType format for taints")
+			return nil, fmt.Errorf("Expected key=value:scheduleType format for taints. Got '%s'", taint)
 		}
-		tokens := strings.FieldsFunc(taint, Split)
-		taintBuilders = append(taintBuilders, cmv1.NewTaint().Key(tokens[0]).Value(tokens[1]).Effect(tokens[2]))
+		// First split effect
+		splitEffect := strings.Split(taint, ":")
+		// Then split key and value
+		split2 := strings.Split(splitEffect[0], "=")
```

Comment:
NP: I suggest to name it something more meaningful then `split2`

Reply:
thanks, fixed